govcd/extension_test.go

```diff
@@ -24,7 +26,112 @@ func (vcd *TestVCD) Test_GetExternalNetwork(check *C) {
 	check.Assert(err, IsNil)
 	LogExternalNetwork(*externalNetwork)
 	check.Assert(externalNetwork.HREF, Not(Equals), "")
-	expectedType := "application/vnd.vmware.admin.extension.network+xml"
 	check.Assert(externalNetwork.Name, Equals, networkName)
-	check.Assert(externalNetwork.Type, Equals, expectedType)
+	check.Assert(externalNetwork.Type, Equals, types.MimeExtensionNetwork)
+}
+
+func (vcd *TestVCD) Test_CreateExternalNetwork(check *C) {
+	if vcd.skipAdminTests {
+		check.Skip("Configuration org != 'System'")
+	}
+
+	virtualCenters, err := QueryVirtualCenters(vcd.client, fmt.Sprintf("(name==%s)", vcd.config.VCD.VimServer))
+	check.Assert(err, IsNil)
+	if len(virtualCenters) == 0 {
+		check.Skip(fmt.Sprintf("No vSphere server found with name '%s'", vcd.config.VCD.VimServer))
+	}
+	vimServerHref := virtualCenters[0].HREF
+
+	// Resolve port group info
+	portGroups, err := QueryPortGroups(vcd.client, fmt.Sprintf("(name==%s;portgroupType==%s)", url.QueryEscape(vcd.config.VCD.ExternalNetworkPortGroup), vcd.config.VCD.ExternalNetworkPortGroupType))
+	check.Assert(err, IsNil)
+	if len(portGroups) == 0 {
+		check.Skip(fmt.Sprintf("No port group found with name '%s'", vcd.config.VCD.ExternalNetworkPortGroup))
+	}
+	if len(portGroups) > 1 {
+		check.Skip(fmt.Sprintf("More then one found with name '%s'", vcd.config.VCD.ExternalNetworkPortGroup))
+	}
+
+	externalNetwork := &types.ExternalNetwork{
+		Name:        TestCreateExternalNetwork,
+		Description: "Test Create External Network",
+		Xmlns:       types.XMLNamespaceExtension,
+		XmlnsVCloud: types.XMLNamespaceVCloud,
+		Configuration: &types.NetworkConfiguration{
+			Xmlns: types.XMLNamespaceVCloud,
+			IPScopes: &types.IPScopes{
+				IPScope: []*types.IPScope{&types.IPScope{
+					Gateway:   "192.168.201.1",
+					Netmask:   "255.255.255.0",
+					DNS1:      "192.168.202.253",
+					DNS2:      "192.168.202.254",
+					DNSSuffix: "some.net",
+					IPRanges: &types.IPRanges{
+						IPRange: []*types.IPRange{
+							&types.IPRange{
+								StartAddress: "192.168.201.3",
+								EndAddress:   "192.168.201.250",
+							},
+						},
+					},
+				},
+				}},
+			FenceMode: "isolated",
+		},
+		VimPortGroupRefs: &types.VimObjectRefs{
+			VimObjectRef: []*types.VimObjectRef{
+				&types.VimObjectRef{
+					VimServerRef: &types.Reference{
+						HREF: vimServerHref,
+					},
+					MoRef:         portGroups[0].MoRef,
+					VimObjectType: vcd.config.VCD.ExternalNetworkPortGroupType,
+				},
+			},
+		},
+	}
+	task, err := CreateExternalNetwork(vcd.client, externalNetwork)
+	check.Assert(err, IsNil)
+	check.Assert(task, Not(Equals), Task{})
```

Comment:
To avoid the triggering data race discussed above (because http.Client does some cleanups inside) I'd suggest to change this line to such:
```suggestion
	check.Assert(task.Task, Not(Equals), types.Task{})
```

Reply:
Tx. Added